add.go

```diff
@@ -518,6 +518,11 @@ func (b *Builder) Add(destination string, extract bool, options AddAndCopyOption
 				go func() {
 					var cloneDir, subdir string
 					cloneDir, subdir, getErr = define.TempDirForURL(tmpdir.GetTempDir(), "", src)
+					defer wg.Done()
+					defer pipeWriter.Close()
```

Comment:
The defer placement feels wrong, doing this after the function call but before errors checking is odd.
IMO they should be at the top of the function.